test/gcloud/dao-integration.js

```diff
@@ -0,0 +1,318 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+var env = require('process').env;
+
+describe('DatastoreDAO', function() {
+  var clearCDS = com.google.cloud.datastore.clear;
+  var DatastoreDAO;
+  var ctx;
+  beforeEach(function() {
+    DatastoreDAO = foam.lookup('org.chromium.apis.web.DatastoreDAO');
+    ctx = foam.__context__.createSubContext({
+      datastore: require('@google-cloud/datastore')(),
+    });
+  });
+
+  var daoFactory = function(cls, opt_ctx) {
+    return clearCDS().then(function() {
+      return DatastoreDAO.create({
+            of: cls,
+            projectId: env.CDS_PROJECT_ID
+          }, opt_ctx || ctx);
+    });
+  };
+
+  // From foam2's test/helpers/generic_dao.js.
+  global.genericDAOTestBattery(daoFactory);
+
+  describe('construction', function() {
```

Comment:
How much of the stuff below is testing the gRPC implementation? Testing the base class would probably be better done closer to the base class.